common/scripts/babel-conditional-preprocess.js

```diff
@@ -0,0 +1,80 @@
+Object.defineProperty(exports, '__esModule', { value: true });
+
+const helper = require('@babel/helper-plugin-utils');
+const t = require('@babel/types');
+exports.default = helper.declare((_api, opts) => {
+  const {
+    annotations
+  } = opts;
+
+  function Handle(path) {
+    let { node } = path;
+    let removed = false;
+
+    for (const annotation of annotations) {
+      const {
+        match
+      } = annotation;
+
+      if (node.leadingComments && node.leadingComments.length > 0) {
```

Comment:
If possible, link to documentation for Node type that shows `leadingComment` always exists.